system/src/system_power_manager.cpp

```diff
@@ -176,14 +190,16 @@ void PowerManager::handleUpdate() {
   // consecutively. The 1st reads fault register status from the last read and the 2nd
   // reads the current fault register status.
   volatile uint8_t lastFault = power.getFault();
-  (void)lastFault;
-  const uint8_t curFault = power.getFault();
+  const uint8_t curFault = power.getFault() | lastFault;
 
   // Watchdog fault or buck converter got disabled
   if ((curFault & 0x80) || !power.isBuckEnabled()) {
```

Comment:
Nit: magic number.

Reply:
Already there.  I'll add a comment for this case.